pkg/security/probe/constantfetch/runtime_compiled.go

```diff
@@ -181,16 +179,18 @@ func (p *constantFetcherRCProvider) GetInputReader(config *ebpf.Config, tm *runt
 	return strings.NewReader(p.cCode), nil
 }
 
-func (a *constantFetcherRCProvider) GetOutputFilePath(config *ebpf.Config, kernelVersion kernel.Version, flagHash string, tm *runtime.RuntimeCompilationTelemetry) (string, error) {
-	hasher := sha256.New()
-	if _, err := hasher.Write([]byte(a.cCode)); err != nil {
+func (a *constantFetcherRCProvider) GetOutputFilePath(config *ebpf.Config, uname *unix.Utsname, flagHash string, tm *runtime.RuntimeCompilationTelemetry) (string, error) {
+	cCodeHash, err := runtime.Sha256hex([]byte(a.cCode))
+	if err != nil {
+		return "", err
+	}
+
+	unameRHash, err := runtime.Sha256hex(uname.Release[:])
```

Comment:
We should probably remove the bytes after the NULL terminator, because we don't know if those bytes are guaranteed to be zero/NULL.